### ISSUE TITLE
Make @Singular compatible with EnumSet

### ISSUE THREAD

**Issue by lsoares** [rzwitserloot/lombok#1189]
As `Set` is supported by @Singular annotation, I suggest `EnumSet `to be as well.
![image](https://cloud.githubusercontent.com/assets/674045/18371928/3fb77636-7630-11e6-8c11-b46e6597f0cd.png)
Do you agree? thx

**Comment by juergenzimmermann** [comment]
Are there any new on this issue?

**Comment by danilovEY** [comment]
@lsoares, why mark @Singular emum fields?
Docs lombok:

> @Singular can only be applied to collection types known to lombok. Currently, the supported types are

**Comment by lsoarestd** [comment]
@danilovEY why not? It would be useful to generate a singular method to add enum values to the annotated list.

**Comment by danilovEY** [comment]
@lsoares, Hm... Although yes, it will be very useful to add support EnumSet. Just a lot of people use List or HashSet for enum type but this is a bad. Interesting, if lombok support Set collection why not supported EnumSet (if EnumSet implement Set interface)?

**Comment by lsoarestd** [comment]
Indeed. EnumSet is optimized for enums. But if nobody uses it, it may not be worth to add more code to Lombok.

**Comment by danilovEY** [comment]
I think it is not hard add this feature. Just add in handle of Singular annotation where check of null field and edit code when create instance object for Set type (replace example like this: EnumSet.noneOf(T.class))

**Comment by lsoarestd** [comment]
great. let's hope they'll do it. or are you proposing that I do a PR?

**Comment by danilovEY** [comment]
If you can do this, will try and may be they accept your pull request :) Because tasks they have and so many.

**Comment by hwaite** [comment]
EnumMap would be helpful as well.
